app/views/users/index.html.haml

```diff
@@ -1,20 +1,88 @@
-.text-center
-  %h1 Users
-%br/
-
-.container
-  .row
-    - @users.each do |user|
-      .col-sm-6.col-md-4.col-lg-3.col-xl-3
-        .card.border-secondary
-          .card-body
-            %h5.card-title
-              = link_to "#{user.first_name}", user
-            %h6.card-subtitle.mb-2.text-muted
-              = user.email
-            = link_to 'Edit', edit_user_path(user), class:'btn btn-primary'
-            = link_to 'Delete', user, method: :delete, data: { confirm: 'Are you sure?' }, class:'btn btn-danger'
-%br/
-%br/
-.text-center
-  = link_to 'New User', new_user_path, class: 'btn btn-info'
\ No newline at end of file
+= form_for :filters, url: users_path, method: 'get' do |filter_form|
+  = filter_form.label :sort_by, 'Sort by'
+  = filter_form.select :sort_by, options_for_select({ '': '',
+                                              'id': 'none',
+                                              'Age': :age,
+                                              'City': :city,
+                                              'First name': :first_name,
+                                              'Last name': :last_name  }, @sort_by)
+  %br/
+  = filter_form.label :male, 'Male'
+  = filter_form.check_box :sex, {:multiple => true, checked: @sex_filter.include?('male')}, 'male', nil
+  %br/
+  = filter_form.label :female, 'Female'
+  = filter_form.check_box :sex, {:multiple => true, checked: @sex_filter.include?('female')}, 'female', nil
+  %br/
+  = filter_form.label :other, 'Other'
+  = filter_form.check_box :sex, {:multiple => true, checked: @sex_filter.include?('other')}, 'other', nil
```

Comment:
I want to show you a good practice of how to refactor this code. Let's discuss later today.